invenio_app_ils/records/views.py

```diff
@@ -18,8 +18,53 @@
 
 from invenio_app_ils.circulation.views import need_permissions
 from invenio_app_ils.errors import DocumentRequestError
-from invenio_app_ils.pidstore.pids import DOCUMENT_REQUEST_PID_TYPE
+from invenio_app_ils.pidstore.pids import DOCUMENT_PID_TYPE, \
+    DOCUMENT_REQUEST_PID_TYPE
 from invenio_app_ils.proxies import current_app_ils_extension
+from invenio_app_ils.signals import record_viewed
+
+
+def create_record_stats_blueprint(app):
+    """Add record stats endpoint to the blueprint."""
+    blueprint = Blueprint(
+        "invenio_app_ils_record_stats", __name__, url_prefix=""
+    )
+    endpoints = app.config.get("RECORDS_REST_ENDPOINTS", [])
+    options = endpoints.get(DOCUMENT_PID_TYPE, {})
+    default_media_type = options.get("default_media_type", "")
+    rec_serializers = options.get("record_serializers", {})
+    serializers = {
+        mime: obj_or_import_string(func)
+        for mime, func in rec_serializers.items()
+    }
+    blueprint.add_url_rule(
+        "{0}/stats/record-view".format(options['item_route']),
```

Comment:
Will there be more types of stats recorded other than views? If so, perhaps it might be quicker to have one API endpoint for recording stats and just passing the name of the event `record-view` with the `POST` request and handling it in the same class but different methods.